lib/rack/mock_response.rb

```diff
@@ -83,13 +83,13 @@ def parse_cookies_from_header
         Array(set_cookie_header).each do |cookie|
           cookie_name, cookie_filling = cookie.split('=', 2)
           cookie_attributes = identify_cookie_attributes cookie_filling
-          parsed_cookie = CGI::Cookie.new(
-            'name' => cookie_name.strip,
-            'value' => cookie_attributes.fetch('value'),
-            'path' => cookie_attributes.fetch('path', nil),
-            'domain' => cookie_attributes.fetch('domain', nil),
-            'expires' => cookie_attributes.fetch('expires', nil),
-            'secure' => cookie_attributes.fetch('secure', false)
+          parsed_cookie = MockCookie.new(
```

Comment:
For compatibility, I think we should use `CGI::Cookie` on Ruby < 3.5, and the new cookie implementation on Ruby 3.5+.

Reply:
Yes, good idea. I've done as much